HHS/Head-Start-TTADP
# Support TTA plan goals import from another region and dynamic ids

**Description of change**
Refactors change from #77 to support dynamic ids as well as import from another region. Included "Grantee Name" as the expected column name in addition to "Grantee (distinct by row number)". Added test for a sample data from region 9.

**How to test**
Running locally:

- yarn deps
- yarn db:migrate
- ./node_modules/.bin/babel-node ./tools/importTTAPlanGoals.js -f R9GranteeTTAPlanTest.csv
- verify that the data has been imported

To test using the original smartsheet file:

- Navigate to https://app.smartsheetgov.com/sheets/fjfHfQCg6mmvRP5ghV77wP9wQ5xg2P2H37QhF7Q1?view=grid
- Export the file to MS Excel or Google Sheets
- Replace smart quotes with non-smart quotes
- Replace "Anticipated Outcome" with "Goal" in the column names
- Save as comma delimited CSV, e.g. R9GranteeTTAPlan.csv
- Follow the above steps substituting the file name with R9GranteeTTAPlan.csv

**Issue(s)**
* closes https://github.com/HHS/Head-Start-TTADP/issues/208

**Checklist**

- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
